framework/src/media/FileOutputDataSource.cpp

```diff
@@ -60,16 +61,35 @@ FileOutputDataSource& FileOutputDataSource::operator=(const FileOutputDataSource
 bool FileOutputDataSource::open()
 {
 	if (!mFp) {
-		setAudioType(utils::getAudioTypeFromPath(mDataPath));
+		audio_type_t audioType;
 
 		mFp = fopen(mDataPath.c_str(), "wb");
-		if (mFp) {
-			medvdbg("file open success\n");
-			return true;
-		} else {
+		if (!mFp) {
 			meddbg("file open failed error : %d\n", errno);
 			return false;
 		}
+
```

Comment:
all these code can be moved to mediautils.
How about createWavHeader?

Reply:
yes. I modified.